timemachine/fe/utils.py

```diff
@@ -485,6 +485,12 @@ def read_sdf(fname: Union[str, Path], removeHs: bool = False) -> List[Chem.Mol]:
     return mols
 
 
+def read_sdf_mols_by_name(fname: str | Path, removeHs: bool = False) -> Dict[str, Chem.Mol]:
```

Comment:
nit: dict is preferred over Dict (deprecated in 3.9) per https://docs.python.org/3/library/typing.html#typing.Dict

```suggestion
def read_sdf_mols_by_name(fname: str | Path, removeHs: bool = False) -> dict[str, Chem.Mol]:
```

Reply:
Good catch on the deprecation. I might prefer to leave this as-is for now though for consistency, in favor of fixing all of the deprecations in a separate PR (e.g. it looks like `List`, `Tuple`, and `Set` are also deprecated)